src/Http/Middleware/RateLimitMiddleware.php

```diff
@@ -251,10 +305,15 @@ protected function getClientIp(ServerRequestInterface $request): string
      */
     protected function getUserIdentifier(ServerRequestInterface $request): string
     {
-        /** @var object|null $user */
+        /**
+         * @var \Authentication\IdentityInterface|\Cake\ORM\Entity|null $user
+         * @phpstan-ignore-next-line
+         */
         $user = $request->getAttribute('identity');
         if ($user) {
-            if (method_exists($user, 'getIdentifier')) {
+            /** @phpstan-ignore-next-line */
+            if ($user instanceof IdentityInterface) {
```

Comment:
You probably won't need the phpstan ignore if you use a interface_exists() check before/along with the instanceof check.

Reply:
Good call.